src/lib/analytics.js

```diff
@@ -138,25 +138,64 @@ store.subscribe(({isSignedIn}) => {
   ga('set', dimensions.SIGNED_IN, isSignedIn ? '1' : '0');
 });
 
-/**
- * Add a listener to detect back/forward cache restores and track them
- * as pageviews with the "bfcache" navigation type set (in case we need
- * to distinguish them from regular pageviews).
- * https://web.dev/bfcache/#how-bfcache-affects-analytics-and-performance-measurement
- */
-window.addEventListener(
-  'pageshow',
+function sendPageView() {
   /**
-   * @param {PageTransitionEvent} e
+   * Add a listener to detect back/forward cache restores and track them
+   * as pageviews with the "back-forward-cache" navigation type set (in
+   * case we need to distinguish them from regular pageviews).
+   * https://web.dev/bfcache/#how-bfcache-affects-analytics-and-performance-measurement
    */
-  (e) => {
-    if (e.persisted) {
-      ga('set', dimensions.NAVIGATION_TYPE, 'back-forward-cache');
-      ga('send', 'pageview');
-    }
-  },
-);
+  window.addEventListener(
+    'pageshow',
+    /**
+     * @param {PageTransitionEvent} e
+     */
+    (e) => {
+      if (e.persisted) {
+        ga('set', dimensions.NAVIGATION_TYPE, 'back-forward-cache');
+        ga('send', 'pageview');
+      }
+    },
+  );
+
+  // Handle prerenders that have already activated
+  if (
```

Comment:
I'd extract all the initial navigation type logic into a single function. Something like:

```js
function getNavigationType() {
  if (document.wasDiscarded) {
    return 'restore';
  }

  const navEntry = self.performance &&
      performance.getEntriesByType &&
      performance.getEntriesByType('navigation')[0];

  if (navEntry) {
    if (navEntry.activationStart > 0) {
      return 'prerender';
    } else {
      return navEntry.type.replace(/_/, '-')
    }
  }
  return '(not set)';
}
```

Then you can do something like this, and keep all your custom dimension setting code together int the logic flow.

```js
ga('set', dimensions.NAVIGATION_TYPE, getNavigationType());
```